### PR TITLE
Remove deprecated automatic logging of gpu metrics

### PR DESCRIPTION
**Pull request by puhuk**
To resolve issue #12521

## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
:)
Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed the deprecated `terminate_on_nan` argument from the `Trainer` constructor ([#12553](https://github.com/PyTorchLightning/pytorch-lightning/pull/12553))
 
 
--
+- Removed the deprecated `gpu_metrics` argument from the `LoggerConnector` constructor ([#12657](https://github.com/PyTorchLightning/pytorch-lightning/pull/12657))
```
Comment (carmocca) [inline CHANGELOG.md]: You missed removing `Trainer(log_gpu_memory)` which is what we want to mention here in the CHANGELOG.

Pushed the removal for you

File: pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py [hunk 2]
```diff
@@ -298,17 +270,6 @@ def metrics(self) -> _METRICS:
...
         return self.trainer._results.metrics(on_step)
 
-    @property
-    def gpus_metrics(self) -> Dict[str, float]:
-        """
-        .. deprecated:: v1.5
-            Will be removed in v1.7.
-        """
-        if isinstance(self.trainer.accelerator, GPUAccelerator) and self.log_gpu_memory:
-            mem_map = memory.get_memory_profile(self.log_gpu_memory)
```
Comment (carmocca) [inline pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:308]: We also need to remove `get_memory_profile`, but you can do that in a separate PR

File: CHANGELOG.md (hunk 1 above)
Comment (puhuk) [reply CHANGELOG.md]: Thank you!

File: pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py (hunk 2 above)
Comment (puhuk) [reply pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:308]: Yes, will send another one.

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
